### PR TITLE
Add temporary pin for PySide6 version

### PR DESCRIPTION
**Pull request by mdickinson**
This PR temporarily pins the PySide6 version in CI, to give us breathing space to adapt to the backwards-incompatible Enum-related changes introduced in PySide6 6.4.0.

### CODE REVIEW

File: etstool.py [hunk 1]
```diff
@@ -214,11 +214,13 @@ def install(edm, runtime, toolkit, environment, editable, source):
     elif toolkit == "pyside6":
         if sys.platform == 'darwin':
             commands.append(
-                "{edm} run -e {environment} -- pip install pyside6<6.2.2'"
+                "{edm} run -e {environment} -- pip install pyside6<6.2.2"
```
Comment (mdickinson) [inline etstool.py:217]: Drive-by fix to remove a spurious quote.

**Comment by mdickinson** [comment]
@corranwebster I'm running into quoting issues on Windows. Any suggestions about how to fix?

**Comment by corranwebster** [comment]
> @corranwebster I'm running into quoting issues on Windows. Any suggestions about how to fix?

No.  I ended up just skipping the EDM Windows/PySide6 combination in CI for the equivalent PR in TraitsUI.

I think it might be fixable by refactoring etstool to pass lists of strings that form a command, rather than a single string, but I didn't have time to do that.

In general the etstools need a bit of a refactor, particularly given the improvements in pip and other related tooling.

**Review by corranwebster — APPROVED** [review]
